src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.SignVerify.cs

```diff
@@ -0,0 +1,179 @@
+            const int Invalid = 0;
+
+            int result = AppleCryptoNative_SecKeyVerifySignature(
+                publicKey,
+                dataHash,
+                signature,
+                hashAlgorithm,
+                signatureAlgorithm,
+                out SafeCFErrorHandle errorHandle);
+
+            using (errorHandle)
+            {
+                return result switch
+                {
+                    Valid => true,
+                    Invalid => false,
+                    kErrorSeeError => throw CreateExceptionForCFError(errorHandle),
+                    kPlatformNotSupported => throw new PlatformNotSupportedException(),
+                    _ => throw new CryptographicException { HResult = result }
+                };
+            }
+        }
+
+        internal static byte[] CreateSignature(
+            SafeSecKeyRefHandle privateKey,
+            ReadOnlySpan<byte> dataHash,
+            PAL_HashAlgorithm hashAlgorithm,
+            PAL_SignatureAlgorithm signatureAlgorithm)
+        {
+            int result = AppleCryptoNative_SecKeyCreateSignature(
+                privateKey,
+                dataHash,
+                hashAlgorithm,
+                signatureAlgorithm,
+                out SafeCFDataHandle signature,
+                out SafeCFErrorHandle errorHandle);
+
+            using (errorHandle)
+            using (signature)
+            {
+                return result switch
+                {
+                    kSuccess => CoreFoundation.CFGetData(signature),
+                    kErrorSeeError => throw CreateExceptionForCFError(errorHandle),
+                    kPlatformNotSupported => throw new PlatformNotSupportedException(),
+                    _ => throw new CryptographicException { HResult = result }
+                };
+            }
+        }
+
+        internal static bool TryCreateSignature(
+            SafeSecKeyRefHandle privateKey,
+            ReadOnlySpan<byte> dataHash,
+            Span<byte> destination,
+            PAL_HashAlgorithm hashAlgorithm,
+            PAL_SignatureAlgorithm signatureAlgorithm,
+            out int bytesWritten)
+        {
+            int result = AppleCryptoNative_SecKeyCreateSignature(
+                privateKey,
+                dataHash,
+                hashAlgorithm,
+                signatureAlgorithm,
+                out SafeCFDataHandle signature,
+                out SafeCFErrorHandle errorHandle);
+
+            using (errorHandle)
+            using (signature)
+            {
+                return result switch
+                {
+                    kSuccess => CoreFoundation.TryCFWriteData(signature, destination, out bytesWritten),
+                    kErrorSeeError => throw CreateExceptionForCFError(errorHandle),
+                    kPlatformNotSupported => throw new PlatformNotSupportedException(),
+                    _ => throw new CryptographicException { HResult = result }
+                };
+            }
```

Comment:
The bulk of these two methods is the same, and involves the same error code checking.  How about

```C#
        private static SafeCFDataHandle NativeCreateSignature(...)
        {
            int result = AppleCryptoNative_SecKeyCreateSignature(
                privateKey,
                dataHash,
                hashAlgorithm,
                signatureAlgorithm,
                out SafeCFDataHandle signature,
                out SafeCFErrorHandle errorHandle);

            using (errorHandle)
            {
                if (result == kSuccess)
                {
                    return signature;
                }

                signature.Dispose();

                // be the right kind of error
            }
        }

        internal static byte[] CreateSignature(...)
        {
            using (SafeCFDataHandle signature = NativeCreateSignature(...))
            {
                return CoreFoundation.CFGetData(signature);
            }
        }

        etc
```